client.go

```diff
@@ -53,6 +66,40 @@ func Timeout(timeout time.Duration) func(*Client) error {
 	}
 }
 
+// Keepalive keeps the connection open.
+func Keepalive() func(*Client) error {
+	return func(c *Client) error {
+		go func(c *Client) {
+			for c.IsConnected() {
+				time.Sleep(keepaliveInterval)
+
+				if err := c.setDeadline(); err != nil {
+					break
+				}
+
+				if _, err := c.conn.Write([]byte(" \n")); err != nil {
+					break
+				}
+
+				if err := c.clearDeadline(); err != nil {
+					break
+				}
+			}
+
+			c.setDisconnected()
+		}(c)
+		return nil
+	}
+}
+
+// NotificationBuffer sets the notification buffer size.
+func NotificationBuffer(size int) func(*Client) error {
+	return func(c *Client) error {
+		c.notify = make(chan Notification, size)
```

Comment:
If this just set a buffer size, which was initially set to the default you can avoid recreating the buffer.

Reply:
How would I solve this in a more elegant way?
Should I check whether c.notify is nil after the Options where called and then set the default one?